src/openzaak/js/components/admin/zaaktypen/selectielijst.js

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: EUPL-1.2
+// Copyright (C) 2020 Dimpact
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import useAsync from 'react-use/esm/useAsync';
+
+const PROCESTYPEN_ENDPOINT = '/admin/api/v1/catalogi/selectielijst/procestypen';
+
+
+const ProcestypeOptions = ({ year, initialValue='' }) => {
+    const {loading, error, value: procestypen} = useAsync(
+        async () => {
+            const query = new URLSearchParams({ year });
+            const response = await window.fetch(`${PROCESTYPEN_ENDPOINT}?${query}`, {credentials: 'same-origin'});
+            if (!response.ok) {
+                throw new Error(response);
+            }
+            const choices = await response.json();
+            return choices;
+        },
+        [year],
+    );
+
+    if (error) {
+        console.error(error);
+        return `Error: ${error}`;
+    }
+
+    let options;
+    if (loading) {
+        options = <option value="">Loading...</option>;
+    } else {
+        // TODO: the selected attribute should work via defaultValue, except it doesn't and I can't figure out why not
```

Comment:
I tried this locally, it seems that `defaultValue` expects an object of the same shape as the elements in `options`. Something like this should work

```javascript
let initialOption = {};
if(options.constructor === Array) initialOption = options.find(option => option.key === initialValue)
...
<select defaultValue={initialOption} name="selectielijst_procestype" id="id_selectielijst_procestype">
```

Reply:
nasty react state/props bug, now fixed by forcing a `key` that changes with the state.